akka-docs/src/main/paradox/coordinated-shutdown.md

```diff
@@ -101,6 +101,11 @@ via `kill SIGTERM` signal (`SIGINT` ctrl-c doesn't work). This behavior can be d
 akka.coordinated-shutdown.run-by-jvm-shutdown-hook=off
 ```
 
+Note that if running in Kubernetes, a SIGKILL will be issued after a set amount of time has passed
+since SIGTERM.  By default this time is 30 seconds ([`terminationGracePeriodSeconds`](https://kubernetes.io/docs/concepts/containers/container-lifecycle-hooks/#hook-handler-execution)):
+it may be worth adjusting the Kubernetes configuration or the phase timeouts to make `CoordinatedShutdown`
+more likely to completely exectue before SIGKILL is received.
```

Comment:
would it be worth changing the `default-phase-timeout` to 4 seconds so that our default totalTimeout is less than 30 seconds?

Reply:
Maybe not in `akka-actor`, which would apply to everyone wherever they're running.  But it could be a responsibility of the Akka Management modules which imply k8s is in use set the default-phase-timeout like that.

Reply:
Crazy idea could be to have a mechanism to read the `terminationGracePeriodSeconds` from k8s and set the default timeout to `floor(terminationGracePeriodSeconds / 7).seconds`, but that would require setting the default timeout after starting the actorsystem.

Reply:
My read of `reference.conf` is that we have 11 levels of phases before `before-actor-system-terminate` (two of which have 10s phase timeouts).  So only a `default-phase-timeout` of `1s` would keep this below 30 seconds?

Reply:
On library (akka-management) can't override reference.conf of another library (akka-actor).
My point is that the value of default-phase-timeout was originally set to something arbitrary, and I can't imagine it would break any existing to change it to 4 seconds. We could do that separately for Akka 2.10.0 (and include note about config change in migration guide).

Reply:
Yeah, makes sense to do it for the next release.